source/patterns/@aws-solutions-constructs/aws-lambda-elasticsearch-kibana/test/lambda-elasticsearch-kibana.test.ts

```diff
@@ -146,4 +131,333 @@ test('check override cognito domain name with provided cognito domain name', ()
+    domainName: 'test-domain',
+    esDomainProps,
+    deployVpc: true,
+  });
+
+  expect(stack).toHaveResource("AWS::Elasticsearch::Domain", {
+    ElasticsearchClusterConfig: {
+      DedicatedMasterCount: 3,
+      DedicatedMasterEnabled: true,
+      InstanceCount: 3,
+      ZoneAwarenessConfig: {
+        AvailabilityZoneCount: 3,
+      },
+      ZoneAwarenessEnabled: false,
+    }
+  });
+});
+
+test("Test ES cluster deploy to 2 AZ when user set availabilityZoneCount to 2", () => {
+  const stack = new cdk.Stack(undefined, undefined, {
+    env: { account: "123456789012", region: 'us-east-1' },
+  });
+
+  const esDomainProps = {
+    elasticsearchClusterConfig: {
+      zoneAwarenessConfig: {
+        availabilityZoneCount: 2
+      }
+    }
+  };
+
+  new LambdaToElasticSearchAndKibana(stack, "lambda-elasticsearch-kibana-stack", {
+    lambdaFunctionProps: getDefaultLambdaProps(),
+    domainName: 'test-domain',
+    esDomainProps,
+    deployVpc: true,
+  });
+
+  expect(stack).toHaveResource("AWS::Elasticsearch::Domain", {
+    ElasticsearchClusterConfig: {
+      DedicatedMasterCount: 3,
+      DedicatedMasterEnabled: true,
+      InstanceCount: 3,
+      ZoneAwarenessConfig: {
+        AvailabilityZoneCount: 2,
+      },
+      ZoneAwarenessEnabled: true,
+    }
+  });
+});
+
+test("Test ES cluster deploy to 3 AZs when user using default values for ElasticsearchClusterConfig", () => {
+  const stack = new cdk.Stack(undefined, undefined, {
+    env: { account: "123456789012", region: 'us-east-1' },
+  });
+
+  new LambdaToElasticSearchAndKibana(stack, "lambda-elasticsearch-kibana-stack", {
+    lambdaFunctionProps: getDefaultLambdaProps(),
+    domainName: 'test-domain',
+    deployVpc: true,
+  });
+
+  expect(stack).toHaveResource("AWS::Elasticsearch::Domain", {
+    ElasticsearchClusterConfig: {
+      DedicatedMasterCount: 3,
+      DedicatedMasterEnabled: true,
+      InstanceCount: 3,
+      ZoneAwarenessConfig: {
+        AvailabilityZoneCount: 3,
+      },
+      ZoneAwarenessEnabled: true,
+    }
+  });
+});
+
+test('check for an existing VPC in construct', () => {
+  const stack = new cdk.Stack(undefined, undefined, {
+    env: { account: "123456789012", region: 'us-east-1' },
+  });
+
+  const vpc = defaults.buildVpc(stack, {
+    defaultVpcProps: defaults.DefaultPublicPrivateVpcProps(),
+    constructVpcProps: {
+      enableDnsHostnames: true,
+      enableDnsSupport: true,
+    },
+    userVpcProps: {
+      vpcName: "existing-vpc-test"
+    }
+  });
+
+  const construct = new LambdaToElasticSearchAndKibana(stack, 'test-lambda-elasticsearch-kibana', {
+    lambdaFunctionProps: getDefaultLambdaProps(),
+    domainName: "test",
+    existingVpc: vpc
+  });
+
+  expect(stack).toHaveResourceLike("AWS::EC2::VPC", {
+    Tags: [
+      {
+        Key: "Name",
+        Value: "existing-vpc-test"
+      }
+    ]
+  });
+
+  expect(stack).toHaveResourceLike("AWS::Elasticsearch::Domain", {
```

Comment:
I believe this is the kind of check I'm looking for elsewhere - at least one place where we deployVpc is true and existingVpc is undefined.